### DISCUSSION TITLE
Get text of clicked cell

### DISCUSSION THREAD

**Discussion by manviny** [adamerose/pandasgui#207] [Q&A]
Is there any way to get the text of a clicked cell?

I open a pandasgui from my program and would need to receive the text of a cell clicked in pandasgui

Thanks.

**Comment by adamerose** [comment]
Edit -> Copy, or Ctrl+C

**Reply by manviny** [reply]
Hi Adam 
    I meant  to communicate to my own program returning the cell value something like:
   `self.submitClicked.emit( 'something' )`

Thanks

**Reply by adamerose** [reply]
Ah I see. Unfortunately there's no method to get that directly through the PandasGUI API. But here's a snippet that might work for you (this is basically just the code I use inside the `copy` method of the `DataFrameViewer` class, slightly modified)

```
indexes = gui.store.selected_pgdf.dataframe_viewer.dataView.selectionModel().selection().indexes()
rows = [ix.row() for ix in indexes]
cols = [ix.column() for ix in indexes]
```

That will give you the indexes of the selected rows and columns, then you can zip these into a list of tuples that would represent the (x,y) of each selected cell and use that to access the data in the DataFrame.

**Reply by manviny** [reply]
Thank for such a great tool.
